app-ios/tutanota/Sources/Sharing/SharingStorage.swift

```diff
@@ -0,0 +1,176 @@
+import Foundation
+
+let TUTANOTA_APP_GROUP = "group.de.tutao.tutanota"
+let TUTANOTA_SHARE_SCHEME = "tutashare://"
+
+/// this gets shared to the main app and contains all the info
+/// to create a new mail & the cleanup
+struct SharingInfo: Codable {
+  /// this is the key that the information is stored under in the
+  /// UserDefaults and on disk
+  var timestamp: String
+  /// body text of new mail
+  var text: String
+  /// files to attach
+  var fileUrls: [URL]
+}
+
+/// different errors that can happen when trying to load, convert and write
+/// shared items to the app group storage
+enum SharingError: Error {
+  case failedToLoad
+  case failedToWrite
+  case failedToRead
+}
+
+/// different types of items are shared under different names
+/// ident identifies the type, content the associated data
+enum SharedItem {
+  case fileUrl(ident: String, content: URL?)
```

Comment:
why is content always nullable?

Reply:
because we can't get the content before loading and we need to know what it is to correctly load it

Reply:
it's fixed now